projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/interactions/cartesian-explorer-context-menu/cartesian-explorer-context-menu.component.test.ts

```diff
@@ -0,0 +1,131 @@
+import { CartesianExplorerContextMenuComponent, ContextMenu } from './cartesian-explorer-context-menu.component';
+
+@Component({
+  // tslint:disable-next-line:component-selector
+  selector: 'test-context-menu-content',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `<div class="test-modal-content">Test Component Content</div> `
+})
+class TestComponent {
+  public constructor(
+    public readonly injector: Injector,
+    @Optional() @Inject(POPOVER_DATA) public readonly data: unknown
+  ) {}
+}
+
+describe('Sheet Overlay component', () => {
+  const selectedData: CartesianSelectedData<unknown> = {
+    timeRange: TimeRangeService.toFixedTimeRange(
+      new Date('2021-11-02T05:33:19.288Z'),
+      new Date('2021-11-02T14:30:15.141Z')
+    ),
+    selectedData: [
+      {
+        dataPoint: { timestamp: '2021-11-02T05:40:00.000Z', value: 1477.3599999999983 },
+        context: {
+          data: [
+            { timestamp: '2021-11-02T05:15:00.000Z', value: 774 },
+            { timestamp: '2021-11-02T05:40:00.000Z', value: 1477.3599999999983 },
+            { timestamp: '2021-11-02T12:05:00.000Z', value: 1056.48 }
+          ],
+          units: 'ms',
+          color: '#4b5f77',
+          name: 'p99',
+          type: CartesianSeriesVisualizationType.Column,
+          stacking: false,
+          hide: false
+        },
+        location: { x: 59, y: 31.023400000000215 }
+      },
+      {
+        dataPoint: { timestamp: '2021-11-02T12:05:00.000Z', value: 1056.48 },
+        context: {
+          data: [
+            { timestamp: '2021-11-02T05:15:00.000Z', value: 774 },
+            { timestamp: '2021-11-02T05:40:00.000Z', value: 1477.3599999999983 },
+            { timestamp: '2021-11-02T12:05:00.000Z', value: 1056.48 }
+          ],
+          units: 'ms',
+          color: '#4b5f77',
+          name: 'p99',
+          type: CartesianSeriesVisualizationType.Column,
+          stacking: false,
+          hide: false
+        },
+        location: { x: 138, y: 82.58120000000001 }
+      }
+    ],
+    location: { x: 452, y: 763 }
+  };
+  let spectator: Spectator<CartesianExplorerContextMenuComponent<unknown>>;
+
+  const createHost = createHostFactory({
+    component: CartesianExplorerContextMenuComponent,
+    declarations: [MockComponent(ButtonComponent), MockComponent(DividerComponent), TestComponent],
+    shallow: true,
+    template: `
+  <ht-cartesian-explorer-context-menu>
+  </ht-cartesian-explorer-context-menu>
+    `,
+    providers: [
+      mockProvider(CartesainExplorerNavigationService, {
+        navigateToExplorer: jest.fn()
+      }),
+      mockProvider(TimeRangeService, {
+        toQueryParams: jest.fn()
+      })
+    ]
+  });
+
+  const createConfiguredHost = ({}) =>
+    createHost(undefined, {
+      providers: [
+        {
+          provide: POPOVER_DATA,
+          deps: [Injector],
+          useFactory: (injector: Injector) => ({
```

Comment:
not quite sure what this factory is meant to do, it looks like it produces an injector that holds POPOVER_DATA. Guessing this was copied from the popover container test, not necessary here. just use `mockProvider(POPOVER_DATA, {/*your mock */})`